com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/adapter/handler/EvaluateRequestHandler.java

```diff
@@ -61,18 +59,11 @@ public void handle(Command command, Arguments arguments, Response response, IDeb
             return;
         }
 
-        // This should be false by default(currently true for test).
-        // User will need to explicitly turn it on by configuring launch.json
-        final boolean showStaticVariables = true;
-        // TODO: when vscode protocol support customize settings of value format, showFullyQualifiedNames should be one of the options.
-        boolean showFullyQualifiedNames = true;
+        final boolean showStaticVariables = UserSettings.showStaticVariables;
+
         Map<String, Object> options = context.getVariableFormatter().getDefaultOptions();
-        if (evalArguments.format != null && evalArguments.format.hex) {
-            options.put(NumericFormatter.NUMERIC_FORMAT_OPTION, NumericFormatEnum.HEX);
-        }
-        if (showFullyQualifiedNames) {
-            options.put(SimpleTypeFormatter.QUALIFIED_CLASS_NAME_OPTION, showFullyQualifiedNames);
-        }
+        VariableUtils.applyFormatterOptions(options, evalArguments.format != null && evalArguments.format.hex,
```

Comment:
Do we need to apply the formatter options every time a evaluate request is processed? If the options stay inside VadiableUtils, then we can call it when a user actually changes the settings.

Reply:
The condition for generating options is also influenced by argument, not only user settings.